src/Orchard.Web/Modules/Orchard.Autoroute/Services/HomeAliasService.cs

```diff
@@ -19,21 +24,27 @@ public class HomeAliasService : IHomeAliasService {
             return _aliasService.Get(HomeAlias);
         }
 
-        public int? GetHomePageId() {
-            var homePageRoute = GetHomeRoute();
-            var homePageIdValue = homePageRoute != null && homePageRoute.ContainsKey("id") ? (string)homePageRoute["id"] : default(string);
-            var homePageId = TryParseInt32(homePageIdValue);
-
-            return homePageId;
+        public int? GetHomePageId(VersionOptions version = null) {
+            var homePage = GetHomePage(version);
+            return homePage != null ? homePage.Id : default(int?);
```

Comment:
Thanks. And equally important, it fixes the issue for content items that use a custom route that doesn't involve the "id" route value (such as Blogs which uses "blogId").